handler/instance-started.go

```diff
@@ -18,22 +18,22 @@ type InstanceStarted struct {
 	ctblr      cantabularClient
 	datasets   datasetAPIClient
 	recipes    recipeAPIClient
-	authToken string
-	// Kafka Producer
+	producer   kafkaProducer
 }
 
-func NewInstanceStarted(c cantabularClient, r recipeAPIClient, d datasetAPIClient, t string) *InstanceStarted {
+func NewInstanceStarted(c cantabularClient, r recipeAPIClient, d datasetAPIClient, p kafkaProducer) *InstanceStarted {
 	return &InstanceStarted{
-		ctblr: c,
-		recipes: r,
+		ctblr:    c,
+		recipes:  r,
 		datasets: d,
-		authToken: t,
+		producer: p,
 	}
 }
 
+// Note to self: why pass cfg rathe than have as part of struct?
```

Comment:
I agree it would be nicer to have the config as part of the struct (we should also change the interface in that case)

Reply:
Cool, might that now then so I don't  forget